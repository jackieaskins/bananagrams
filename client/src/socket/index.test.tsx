import { socket, createGame, joinGame, disconnect } from './index';

jest.mock('socket.io-client', () => ({
  io: () => ({
    emit: jest.fn().mockName('emit'),
    disconnect: jest.fn().mockName('disconnect'),
  }),
}));

const mockCallback = jest.fn();
describe('socket', () => {
  test('returns result of io()', () => {
    expect(socket).toMatchSnapshot();
  });

  test('createGame calls emit', () => {
    const props = {
      gameName: 'gameName',
      username: 'username',
      isShortenedGame: true,
    };
    createGame(props, mockCallback);

    expect(socket.emit).toHaveBeenCalledWith('createGame', props, mockCallback);
  });

  test('joinGame calls emit', () => {
    const props = { gameId: 'gameId', username: 'username' };
    joinGame(props, mockCallback);

    expect(socket.emit).toHaveBeenCalledWith('joinGame', props, mockCallback);
  });

  test('disconnect calls disconnect', () => {
    disconnect();

    expect(socket.disconnect).toHaveBeenCalledWith();
  });
});
