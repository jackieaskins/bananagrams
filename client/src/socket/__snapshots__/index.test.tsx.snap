// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`socket returns result of io() 1`] = `
Object {
  "disconnect": [MockFunction disconnect],
  "emit": [MockFunction emit],
}
`;
